Support path or buffer inputs to the x509 verify method

astitt-ripple opened the pull request:
For your consideration, allows for the use case where it is more convenient to use certs and/or CAs from memory than from files.

yorkie commented:
@astitt-ripple As you said before, we should handle the buffer and path in JavaScript land, keep using memory at the C++ bridge code.

astitt-ripple commented:
Any idea why travis-ci fails? I cannot see the logs, and `npm test` is passing in my environment.

Southern commented:
Travis tests are failing for 0.10 and 0.12:
```
/home/travis/build/Southern/node-x509/index.js:68
  if (String(certPathOrString).startsWith('-----BEGIN')) {
                               ^
TypeError: Object /home/travis/build/Southern/node-x509/test/certs/enduser-example.com.crt has no method 'startsWith'
    at Object.exports.verify (/home/travis/build/Southern/node-x509/index.js:68:32)
    at x509_verify_test (/home/travis/build/Southern/node-x509/test/test.js:26:8)
    at Object.<anonymous> (/home/travis/build/Southern/node-x509/test/test.js:32:1)
    at Module._compile (module.js:456:26)
    at Object.Module._extensions..js (module.js:474:10)
    at Module.load (module.js:356:32)
    at Function.Module._load (module.js:312:12)
    at Function.Module.runMain (module.js:497:10)
    at startup (node.js:119:16)
    at node.js:945:3
```

astitt-ripple commented:
Ah I see now, thank you @Southern! I was able to reproduce the error locally, and copied in the polyfill from the mozilla developer portal.

@yorkie I've updated the code to read the certs and CABundle in JS instead, and removed the corresponding C++ code

Southern commented:
@astitt-ripple were you able to get this working? Looks like the tests passed when they were ran last.

astitt-ripple commented:
@Southern I was able to get it working 👍  There were some memory leak issues I found/resolved during testing as well. I can re-open a PR if this is something you'd consider merging.
